### ISSUE TITLE
Errno::ENOENT - No such file or directory @ rb_sysopen

### ISSUE THREAD

**Issue by ahmedk92** [CocoaPods/CocoaPods#9098]
### Command

```
/usr/local/Cellar/cocoapods/1.7.5/libexec/bin/pod install --repo-update
```

### Report

* What did you do?

* What did you expect to happen?

* What happened instead?

### Stack

```
   CocoaPods : 1.7.5
        Ruby : ruby 2.3.7p456 (2018-03-28 revision 63024) [universal.x86_64-darwin18]
    RubyGems : 3.0.4
        Host : Mac OS X 10.14.5 (18F132)
       Xcode : 10.2.1 (10E1001)
         Git : git version 2.20.1 (Apple Git-117)
Ruby lib dir : /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib
Repositories : master - https://github.com/CocoaPods/Specs.git @ 8e533fd5d789782e6a7e295c44b3cf659490e372
               
```

### Plugins

```
cocoapods-deintegrate : 1.0.4
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.1.0
cocoapods-trunk       : 1.3.1
cocoapods-try         : 1.1.0
```
### Error

```
Errno::ENOENT - No such file or directory @ rb_sysopen - /Users/ahmedkhalaf/.cocoapods/repos/visualstudio-defaultcollection-arabia%20pods-_git-arabia%20pods/all_pods_versions.txt
/Library/Ruby/Gems/2.3.0/gems/cocoapods-core-1.7.5/lib/cocoapods-core/cdn_source.rb:276:in `initialize'
/Library/Ruby/Gems/2.3.0/gems/cocoapods-core-1.7.5/lib/cocoapods-core/cdn_source.rb:276:in `open'
/Library/Ruby/Gems/2.3.0/gems/cocoapods-core-1.7.5/lib/cocoapods-core/cdn_source.rb:276:in `local_file'
/Library/Ruby/Gems/2.3.0/gems/cocoapods-core-1.7.5/lib/cocoapods-core/cdn_source.rb:253:in `ensure_versions_file_loaded'
/Library/Ruby/Gems/2.3.0/gems/cocoapods-core-1.7.5/lib/cocoapods-core/cdn_source.rb:187:in `search'
/Library/Ruby/Gems/2.3.0/gems/cocoapods-core-1.7.5/lib/cocoapods-core/source/aggregate.rb:83:in `block in search'
/Library/Ruby/Gems/2.3.0/gems/cocoapods-core-1.7.5/lib/cocoapods-core/source/aggregate.rb:83:in `select'
/Library/Ruby/Gems/2.3.0/gems/cocoapods-core-1.7.5/lib/cocoapods-core/source/aggregate.rb:83:in `search'
/Library/Ruby/Gems/2.3.0/gems/cocoapods-1.7.5/lib/cocoapods/resolver.rb:389:in `create_set_from_sources'
/Library/Ruby/Gems/2.3.0/gems/cocoapods-1.7.5/lib/cocoapods/resolver.rb:358:in `find_cached_set'
/Library/Ruby/Gems/2.3.0/gems/cocoapods-1.7.5/lib/cocoapods/resolver.rb:332:in `specifications_for_dependency'
/Library/Ruby/Gems/2.3.0/gems/cocoapods-1.7.5/lib/cocoapods/resolver.rb:139:in `search_for'
/Library/Ruby/Gems/2.3.0/gems/cocoapods-1.7.5/lib/cocoapods/resolver.rb:248:in `block in sort_dependencies'
/Library/Ruby/Gems/2.3.0/gems/cocoapods-1.7.5/lib/cocoapods/resolver.rb:241:in `each'
/Library/Ruby/Gems/2.3.0/gems/cocoapods-1.7.5/lib/cocoapods/resolver.rb:241:in `sort_by'
/Library/Ruby/Gems/2.3.0/gems/cocoapods-1.7.5/lib/cocoapods/resolver.rb:241:in `sort_dependencies'
/Library/Ruby/Gems/2.3.0/gems/molinillo-0.6.6/lib/molinillo/delegates/specification_provider.rb:53:in `block in sort_dependencies'
/Library/Ruby/Gems/2.3.0/gems/molinillo-0.6.6/lib/molinillo/delegates/specification_provider.rb:70:in `with_no_such_dependency_error_handling'
/Library/Ruby/Gems/2.3.0/gems/molinillo-0.6.6/lib/molinillo/delegates/specification_provider.rb:52:in `sort_dependencies'
/Library/Ruby/Gems/2.3.0/gems/molinillo-0.6.6/lib/molinillo/resolution.rb:288:in `initial_state'
/Library/Ruby/Gems/2.3.0/gems/molinillo-0.6.6/lib/molinillo/resolution.rb:210:in `start_resolution'
/Library/Ruby/Gems/2.3.0/gems/molinillo-0.6.6/lib/molinillo/resolution.rb:168:in `resolve'
/Library/Ruby/Gems/2.3.0/gems/molinillo-0.6.6/lib/molinillo/resolver.rb:43:in `resolve'
/Library/Ruby/Gems/2.3.0/gems/cocoapods-1.7.5/lib/cocoapods/resolver.rb:88:in `resolve'
/Library/Ruby/Gems/2.3.0/gems/cocoapods-1.7.5/lib/cocoapods/installer/analyzer.rb:939:in `block in resolve_dependencies'
/Library/Ruby/Gems/2.3.0/gems/cocoapods-1.7.5/lib/cocoapods/user_interface.rb:64:in `section'
/Library/Ruby/Gems/2.3.0/gems/cocoapods-1.7.5/lib/cocoapods/installer/analyzer.rb:937:in `resolve_dependencies'
/Library/Ruby/Gems/2.3.0/gems/cocoapods-1.7.5/lib/cocoapods/installer/analyzer.rb:118:in `analyze'
/Library/Ruby/Gems/2.3.0/gems/cocoapods-1.7.5/lib/cocoapods/installer.rb:398:in `analyze'
/Library/Ruby/Gems/2.3.0/gems/cocoapods-1.7.5/lib/cocoapods/installer.rb:221:in `block in resolve_dependencies'
/Library/Ruby/Gems/2.3.0/gems/cocoapods-1.7.5/lib/cocoapods/user_interface.rb:64:in `section'
/Library/Ruby/Gems/2.3.0/gems/cocoapods-1.7.5/lib/cocoapods/installer.rb:220:in `resolve_dependencies'
/Library/Ruby/Gems/2.3.0/gems/cocoapods-1.7.5/lib/cocoapods/installer.rb:156:in `install!'
/Library/Ruby/Gems/2.3.0/gems/cocoapods-1.7.5/lib/cocoapods/command/install.rb:51:in `run'
/Library/Ruby/Gems/2.3.0/gems/claide-1.0.2/lib/claide/command.rb:334:in `run'
/Library/Ruby/Gems/2.3.0/gems/cocoapods-1.7.5/lib/cocoapods/command.rb:52:in `run'
/Library/Ruby/Gems/2.3.0/gems/cocoapods-1.7.5/bin/pod:55:in `<top (required)>'
/usr/local/Cellar/cocoapods/1.7.5/libexec/bin/pod:22:in `load'
/usr/local/Cellar/cocoapods/1.7.5/libexec/bin/pod:22:in `<main>'
```

I also tried solutions in similar issues, but unfortunately none worked so far.

All what I tried:

- Deleting Pods directory.
- Deleting .cocoapods repo folder
- pod install --repo-update
- pod cache clean --all
- pod deintegrate
- npm install

Thanks!

**Comment by dnkoutso** [comment]
This seems your repo is acting as a CDN repo where maybe it shouldnt? cc @igor-makarov

**Comment by dnkoutso** [comment]
`pod repo list` and `pod repo remove <bad_repo_name>` to unblock yourself.

**Comment by igor-makarov** [comment]
Judging by the logs, this is some sort of Visual Studio CP repo mirror. Is that so, @ahmedk92? 

If so, is VS creating an `.url` file in the repo dir? That is part of the `CDNSource` implementation detail and might be causing the `Source::Manager` to mistake the repo for a CDN.

@dnkoutso Here's the relevant code:
* [1-7-stable version](https://github.com/CocoaPods/Core/blob/1-7-stable/lib/cocoapods-core/source/manager.rb#L293)
* [1-8-stable version](https://github.com/CocoaPods/Core/blob/1-8-stable/lib/cocoapods-core/source/manager.rb#L307)

If this is indeed the case, VS support must be notified.

**Comment by ahmedk92** [comment]
Yes, it's an Azure Repo (Azure DevOps). And it does have `.url` file. 

<img width="328" alt="Screen Shot 2019-08-11 at 2 18 57 PM" src="https://user-images.githubusercontent.com/11476927/62833696-3e908480-bc43-11e9-96c7-e0d485d72d51.png">

**Comment by igor-makarov** [comment]
I'm guessing that URL is set by Azure?

**Comment by ahmedk92** [comment]
> I'm guessing that URL is set by Azure?

No idea what this means 😅

> How was the repo added?

In the top of my Podfile I have:

source 'https://github.com/CocoaPods/Specs.git'
source 'link to the private repo'

**Comment by igor-makarov** [comment]
Ok @ahmedk92 I've figured out your problem. This is a problem in the logic that determines whether a source URL is a CDN or not and was resolved by #9006.

You can update to the `1.8.0.beta1` version of CP where this shouldn't happen.

**Comment by ahmedk92** [comment]
Thank you so much! It's working now.
